src/2dplane/ObstacleGrid.cpp

```diff
@@ -24,6 +25,25 @@ Vector2i ObstacleGrid::gridSquareForLocation(const Vector2f &loc) const {
                     loc.y() / height() * discretizedHeight());
 }
 
+float ObstacleGrid::nearestObstacleDist(const Vector2f &state, float maxDist) const {
```

Comment:
I think you need to convert `maxDist` into discrete coordinates here, similar to how you scale `state` into `x` and `y`.